src/basic/fileio.c

```diff
@@ -148,6 +148,28 @@ int write_string_stream_ts(
                         return -EBADF;
         }
 
+        if (flags & WRITE_STRING_VIRTUALFILE_SUPRESS_IF_POSSIBLE) {
+                _cleanup_free_ char *ret = NULL;
+
+                if (fd < 0) {
+                        fd = fileno(f);
+                        if (fd < 0)
+                                return -EBADF;
+                }
+
+                r = read_virtual_stream(fd, SIZE_MAX, &ret, NULL);
```

Comment:
There were couple of options available to read the procfs:
- Use the existing `read_virtual_file` (modified as `read_virtual_stream` to take the file descriptor) which seems to error tested to read proc fs. However, I am not sure if reading a `*_VIRTUALFILE_*` sits well within the abstraction of generic write function, `write_string_stream_ts`.
- Use an approach similar to `verify_file`, which caters to the similar use case of swallowing an error if the file contents are the same.

I used option 1 but lmk what feels right.

Reply:
i think the approach you took on this is Ok